### ISSUE TITLE
merging branch when tag with same name exists

### ISSUE THREAD

**Issue by Deeem2031** [fork-dev/TrackerWin#1873]
1. Fork 1.83.1.0
2. create tag "test"
3. create branch "test" and add at least one commit to it
4. try to merge the branch "test" with default merge option

### expected
fork merges commits of the branch

### observed
nothing, at least that's what seems to happen at first. The merge window closes and you get no changes. What does happen in the background is fork merges the tag which is at the same commit thus creating no changes.

### references
I found this behavior referenced in [issue 570](https://github.com/fork-dev/TrackerWin/issues/570#issuecomment-559571139_)

> In order to solve conflict with same named branch and tag, we replaced `git merge branch` with `git merge /refs/heads/branch`.

Also kind of similar is [Issue 744: Cannot apply stash when there is branch "stash"](https://github.com/fork-dev/TrackerWin/issues/744)

### ideas
As the comment in 570 suggests it can't be fixed with fully qualifying all branches without sacrificing proper merge messages. Maybe only fully qualify the name when there is ambiguity (dunno if there is a way to detect that in advance) or at least show the warning the git command **does** output (warning: refname 'test' is ambiguous.) and inform the user.

**Comment by DanPristupov** [comment]
Yes, the commit message is the reason why we can't use full branch name on merge `git merge refs/heads/test` https://github.com/fork-dev/TrackerWin/issues/570#issuecomment-559571139.

Isn't that a minor issue though?

**Comment by Deeem2031** [comment]
In hindsight this might seem very minor. The real problem is discovering the tag (which in my real world case was created at least 2 years ago by some one else) without any hint coming from fork. It took me more than just a few moments and the use of the console to realize the tag existed. I can imagine someone without knowledge to use the console (which we have enough in our company) spending hours trying to figure out what is happening.

At least I don't see no downside to displaying warnings git commands may output.

**Comment by DanPristupov** [comment]
Makes sense. Thank you for the explanation.

**Comment by ChettoZzz** [comment]
Two years later, and... it happened to me just today.
I remember several times invoking a command from Fork without anything happening. 
Only by re-running the git command from the console I got any clue.
This time, it was due to ambiguity between a branch and a tag. Another time, it was an issue with LFS.
It's really frustrating that nothing happens without any feedback or warning being displayed.
Getting just "everything is up to date" is misleading when there’s actually an issue.
